homeassistant/components/venstar/sensor.py

```diff
@@ -0,0 +1,135 @@
+"""Representation of Venstar sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    STATE_CLASS_MEASUREMENT,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, TEMP_CELSIUS, TEMP_FAHRENHEIT
+from homeassistant.helpers.entity import Entity
+
+from . import VenstarDataUpdateCoordinator, VenstarEntity
+from .const import DOMAIN
+
+
+@dataclass
+class VenstarSensorTypeMixin:
+    """Mixin for sensor required keys."""
+
+    cls: type[VenstarSensor]
+    stype: str
+
+
+class VenstarSensor(VenstarEntity, SensorEntity):
+    """Base class for a Venstar sensor."""
+
+    entity_description: VenstarSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: VenstarDataUpdateCoordinator,
+        config: ConfigEntry,
+        entity_description: VenstarSensorEntityDescription,
+        sensor_name: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, config)
+        self.entity_description = entity_description
+        self.sensor_name = sensor_name
+        self._config = config
+
+    @property
+    def unique_id(self):
+        """Return the unique id."""
+        return f"{self._config.entry_id}_{self.sensor_name.replace(' ', '_')}_{self.entity_description.key}"
+
+
+class VenstarHumiditySensor(VenstarSensor):
+    """Represent a Venstar humidity sensor."""
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return f"{self._client.name} {self.sensor_name} Humidity"
+
+    @property
+    def native_value(self) -> int:
+        """Return state of the sensor."""
+        return self._client.get_sensor(self.sensor_name, "hum")
+
+
+class VenstarTemperatureSensor(VenstarSensor):
+    """Represent a Venstar temperature sensor."""
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return (
+            f"{self._client.name} {self.sensor_name.replace(' Temp', '')} Temperature"
+        )
+
+    @property
+    def native_unit_of_measurement(self) -> str:
+        """Return unit of measurement the value is expressed in."""
+        if self._client.tempunits == self._client.TEMPUNITS_F:
+            return TEMP_FAHRENHEIT
+        return TEMP_CELSIUS
+
+    @property
+    def native_value(self) -> float:
+        """Return state of the sensor."""
+        return round(float(self._client.get_sensor(self.sensor_name, "temp")), 1)
+
```

Comment:
Usually we put the entities themselves after `async_setup_entry`

Reply:
I was getting problems with the SENSOR_ENTITIES cls members not having a definition for the classes..  Is there a way to forward declare them like in C?

Reply:
Take a look at Renault sensors I think I had a similar issue

Reply:
You could move `SENSOR_ENTITIES` to the end of the file instead so it defined after the classes

Reply:
I think its just SENSOR_ENTITIES that needs to go to ghe nottom